### ISSUE TITLE
How to see which processor has which gateset

### ISSUE THREAD

**Issue by balopat** [quantumlib/Cirq#4279]
**Is your feature request related to a use case or problem? Please describe.**

It would be great if the processor would "know" what gateset it accepts - instead of having to pass that in. I find this rather odd. While there is a way to list processors using the Engine API or the UI, there is no way to tell what gate set a processor accepts. The `processor.get_device()` method takes a list of gate sets which then it uses to validate operations. 

```
processor = engine.get_processor('rainbow')
# Print the device showing qubit connectivity for the Sycamore gateset.
device = processor.get_device([cg.SYC_GATESET])
```

**Describe the solution you'd like**

Instead it would be great if a gate set could be determined based on a processor ID. 

**Describe alternatives/workarounds you've considered**

Alternative is to ask someone, use documentation/datasheets (if available). 

**What is the urgency from your perspective for this issue? Is it blocking important work?**
P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by dstrain115** [comment]
@verult I think this can be closed as part of the device spec work.

**Comment by verult** [comment]
Yep, closing in favor of https://github.com/quantumlib/Cirq/issues/5050
